### PR TITLE
4063: Setup parent/child version relationships for institutions and zip code rates.

### PR DESCRIPTION
**Pull request by elder-joseph**
## Description
As a developer, I need to add a parent / child relationship so that the database is more normalized.

This PR is specific to the GIDS app source code, there is a separate PR associated with the DB changes for 4063: https://github.com/department-of-veterans-affairs/gibct-data-service/pull/519

https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/4063

## Testing done
Testing passes locally 

## Screenshots
N/A

## Acceptance criteria
- [x] Version table is created.
- [x] Has_many relationship is established from version to institution.
- [x] Has_many relationship is established from version to zipcode_rate.
- [x] Archiver is configured to archive institution and zipcode_rate data.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/models/institution_builder.rb [hunk 1]
```diff
@@ -89,13 +89,15 @@ def self.initialize_with_weams(version_number)
...
-
+    str = "INSERT INTO institutions (#{columns.join(', ')}, version, created_at, updated_at, version_id) "
+    str += Weam
+           .select(columns)
+           .select("#{version_number.to_i} as version")
+           .select("#{conn.quote(timestamp)} as created_at")
+           .select("#{conn.quote(timestamp)} as updated_at")
+           .select('v.id as version_id')
+           .to_sql
+    str += " JOIN versions v ON v.number = #{version_number}"
```
Comment (shawkey-daniel) [inline app/models/institution_builder.rb]: Please make this INNER JOIN for clarity.

File: app/models/institution_builder.rb [hunk 2]
```diff
@@ -482,16 +485,17 @@ def self.build_zip_code_rates_from_weams(version_number)
       concat_ws(', ', physical_city, physical_state) as physical_location,
       ?,
```
Comment (shawkey-daniel) [inline app/models/institution_builder.rb]: Change to v.number and pass version_number once.

File: app/models/institution_builder.rb [hunk 3]
```diff
@@ -482,16 +485,17 @@ def self.build_zip_code_rates_from_weams(version_number)
...
       #{conn.quote(timestamp)},
-      #{conn.quote(timestamp)}
-    FROM weams
+      #{conn.quote(timestamp)},
+      v.id
+      FROM weams JOIN versions v ON v.number = ?
```
Comment (shawkey-daniel) [inline app/models/institution_builder.rb]: INNER JOIN for clarity.

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]
Minor changes.

**Review by sonntag-adam — APPROVED** [review]

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by jholton — APPROVED** [review]

**Pull request merged**